### PR TITLE
Refactor or to avro method

### PR DESCRIPTION
**Pull request by buptyzc**
In our prod environment, we find the orToAvro function is inefficient. So we make a map to cache the relation between the field object and the process function.With this change, we ignore invoking string contains function every time. The process speed improved.

### CODE REVIEW

File: databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+      }
+      return converter;
+    }
+
+    @Override
+    public Object convert(Column s, Field avroField) throws DatabusException
+    {
+      Object obj = s.getValue();
+      Integer res = null;
+      if (obj instanceof Integer)
```
Comment (chavdar) [inline databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java]: nit: You can combine most of those checks into one:

```
if (obj instanceof Number) {
   res = ((Number)obj).intValue()
}
```

File: databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+      {
+        res = Integer.valueOf(((Double) obj).intValue());
+      }
+      else if (obj instanceof Float)
+      {
+        res = Integer.valueOf(((Float) obj).intValue());
+      }
+      else if (obj instanceof byte[])
+      {
+        String resStr = new String((byte[]) obj, Charset.defaultCharset());
```
Comment (chavdar) [inline databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java]: Use the default character set is dangerous. Is the byte[] even possible in this case?

File: databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+    @Override
+    public Object convert(Column s, Field avroField) throws DatabusException
+    {
+      Object obj = s.getValue();
+      Long res = null;
+      if (obj instanceof Date)
+      {
+        return ((Date) obj).getTime();
+      }
+      else if (obj instanceof Long)
```
Comment (chavdar) [inline databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java]: Same comment as before about using Number.

File: databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java (hunk 1 above)
Comment (buptyzc) [inline databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java]: thank you

File: databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java (hunk 2 above)
Comment (buptyzc) [inline databus2-relay/databus2-event-producer-or/src/main/java/com/linkedin/databus2/producers/util/Or2AvroConvert.java]: yes,it's logically imposible.

**Pull request merged**